Doc/#392 - Documentation for Data migration API (#499)
Avaiga/taipy-doc: docs/manuals/core/versioning/index.md

- In development mode (default mode), Taipy drops all old entities before running the application.

>   It is made to help users during the application development phase to implement its application

Reviewer: ```suggestion
  It is made to help users during the application development phase to implement their application
```

>   through successive iterations of configuration changes. Please refer to the

Reviewer: ```suggestion
  through successive iterations of configuration changes. For more information, please refer to
```

>   [Development mode](./development_mode.md) documentation page for more details.

Reviewer: ```suggestion
  [Development mode](./development_mode.md).
```

- In experiment mode, Taipy keeps old entities untouched but filters them out when running the
  application. The application behaves like there are no old entities. Only the entities created
  during the current run are considered by the application. It is designed to help the user improve
  an existing application by experimenting with possible configuration changes, trying new

>   algorithms, investigating the impacts of a parameter change, etc. Please refer to the

Reviewer: ```suggestion
  algorithms, investigating the impacts of a parameter change, etc. For more information, please refer to
```

>   [Experiment mode](./experiment_mode.md) documentation page for more details.

Reviewer: ```suggestion
  [Experiment mode](./experiment_mode.md).
```

- In production mode, Taipy considers all existing entities, whether they have been instantiated in
  the current run or in a previous one. It is designed to run an application in a production

>   environment with existing entities created in previous runs. Please refer to the

Reviewer: ```suggestion
  environment with existing entities created in previous runs. For more information, please refer to
```

>   [Production mode](./production_mode.md) documentation page for more details.

Reviewer: ```suggestion
  [Production mode](./production_mode.md).
```
